src/overlays/actors/ovl_En_Ossan/z_en_ossan.c

```diff
@@ -21,148 +84,1642 @@ const ActorInit En_Ossan_InitVars = {
+            EnOssan_ResetItemPosition(this);
+            item = this->items[this->cursorIdx];
+            item->restockFunc(globalCtx, item);
+            if (!EnOssan_TestEndInteraction(this, globalCtx, CONTROLLER1(globalCtx))) {
+                switch (globalCtx->msgCtx.choiceIndex) {
+                    case 0:
+                        func_8019F208();
+                        player->actor.shape.rot.y = BINANG_ROT180(player->actor.shape.rot.y);
+                        player->stateFlags2 |= 0x20000000;
+                        func_801518B0(globalCtx, this->textId, &this->actor);
+                        EnOssan_SetupStartShopping(globalCtx, this, true);
+                        func_800B85E0(&this->actor, globalCtx, 100.0f, -1);
+                        break;
+                    case 1:
+                    default:
+                        func_8019F230();
+                        EnOssan_EndInteraction(globalCtx, this);
+                        break;
+                }
+            }
+        }
+    } else if (talkState == 5 && func_80147624(globalCtx)) {
+        EnOssan_ResetItemPosition(this);
+        item = this->items[this->cursorIdx];
+        item->restockFunc(globalCtx, item);
+        player->actor.shape.rot.y = BINANG_ROT180(player->actor.shape.rot.y);
+        player->stateFlags2 |= 0x20000000;
+        func_801518B0(globalCtx, this->textId, &this->actor);
+        EnOssan_SetupStartShopping(globalCtx, this, true);
+        func_800B85E0(&this->actor, globalCtx, 100.0f, -1);
+    }
+}
+
+void EnOssan_ItemPurchased(EnOssan* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if (this->cutsceneState == ENOSSAN_CUTSCENESTATE_STOPPED) {
+        if (ActorCutscene_GetCanPlayNext(this->cutscene)) {
+            ActorCutscene_StartAndSetFlag(this->cutscene, &this->actor);
+            player->stateFlags2 |= 0x20000000;
+            EnOssan_SetupAction(this, EnOssan_ContinueShopping);
+            this->cutsceneState = ENOSSAN_CUTSCENESTATE_PLAYING;
+        } else {
+            if (ActorCutscene_GetCurrentIndex() == 0x7C) {
+                ActorCutscene_Stop(0x7C);
+            }
+            this->cutscene = this->lookToShopkeeperCutscene;
+            ActorCutscene_SetIntentToPlay(this->cutscene);
+        }
+    }
+    if (func_800B84D0(&this->actor, globalCtx)) {
+        func_80151938(globalCtx, 0x0642);
+    } else {
+        func_800B85E0(&this->actor, globalCtx, 400.0f, -1);
+    }
+}
+
+void EnOssan_PositionSelectedItem(EnOssan* this) {
+    static Vec3f sSelectedItemPosition[] = {
+        { 35.0f, 68.0f, -130.0f },
+        { -35.0f, 68.0f, -130.0f },
+    };
+
+    EnGirlA* item;
+    u8 i;
+    u8 i2;
+    ShopItem* shopItem;
+    Vec3f worldPos;
+
+    i = this->cursorIdx;
+    shopItem = &sShops[this->actor.params][i];
+    item = this->items[i];
+
+    i2 = i >> 2;
+    VEC3F_LERPIMPDST(&worldPos, &shopItem->spawnPos, &sSelectedItemPosition[i2], this->shopItemSelectedTween);
```

Comment:
```suggestion
    };
    EnGirlA* item;
    u8 i = this->cursorIdx;
    u8 i2;
    ShopItem* shopItem = &sShops[this->actor.params][i];
    Vec3f worldPos;
    
    item = this->items[i];
    i2 = i >> 2;
    VEC3F_LERPIMPDST(&worldPos, &shopItem->spawnPos, &sSelectedItemPosition[i2], this->shopItemSelectedTween);
```
I think I'd prefer something like this. It's variable how you want to do it though